fe/fe-core/src/main/java/com/starrocks/backup/RestoreJob.java

```diff
@@ -527,6 +544,15 @@ private void checkAndPrepareMeta() {
                 Table localTbl = globalStateMgr.getLocalMetastore()
                             .getTable(db.getFullName(), jobInfo.getAliasByOriginNameIfSet(tblInfo.name));
                 if (localTbl != null) {
+                    if (remoteTbl.isOlapView() && !localTbl.isOlapView()) {
+                        status = new Status(ErrCode.BAD_REPLACE,
+                                            "Table: " + tblInfo.name + " has existed and it is not a View");
+                        return;
+                    } else if (remoteTbl.isOlapView()) {
+                        restoredOlapViews.put((View) remoteTbl, true);
+                        continue;
+                    }
```

Comment:
there are 2 `remoteTbl.isOlapView()`, could refactor the code.

Reply:
fixed